sdk/core/dma/dma_compartment.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <cdefs.h>
+#include <stdint.h>
+
+/**
+ * A function below claims the source and target addresses of the DMA interface.
+ * While, DMA is in progress, these addresses should not be freed
```

Comment:
More explicitly: they will be claimed by the DMA compartment and so the memory will not be freed.